DailyReportWeb/Controllers/Api/PolicyController.cs

```diff
@@ -55,27 +42,118 @@ public void Put([FromBody]PolicySummary policySummary)
             using (var db = new ReportsDb())
             {
                 var reportSettings = db.ReportSettings.SingleOrDefault(qr => qr.ProjectId == policySummary.ProjectId && qr.BaseUrl == policySummary.BaseUrl);
-
                 if (reportSettings == null)
                 {
-                    reportSettings = new ReportSettings();
+                    reportSettings = CreateFromPolicySummary(policySummary);
                     db.ReportSettings.Add(reportSettings);
+                }
+                reportSettings.ReportTime = policySummary.ReportTime;
 
-                    reportSettings.BaseUrl = policySummary.BaseUrl;
-                    reportSettings.SharedSecret = policySummary.SharedSecret;
-                    reportSettings.ProjectId = policySummary.ProjectId;
+                db.SaveChanges();
+            }
         }
 
-                reportSettings.ReportTime = policySummary.ReportTime;
+        // PUT: api/Policy/5
+        public void Put(long id, [FromBody]ReportSettings updatedReportSettings)
+        {
+            using (var db = new ReportsDb())
+            {
+                var reportSettings = db.ReportSettings.SingleOrDefault(qr => qr.ProjectId == updatedReportSettings.ProjectId && qr.BaseUrl == updatedReportSettings.BaseUrl);
+                if (reportSettings == null)
+                {
+                    reportSettings = new ReportSettings();
+                    db.ReportSettings.Add(reportSettings);
+                }
+                updatedReportSettings.CopyProperties(reportSettings);
+                reportSettings.PolicyXml = Serialization.XmlSerialize(updatedReportSettings.Policy);
 
                 db.SaveChanges();
             }
         }
 
+        // Policies are not created directly!
+        // POST: api/Policy
+        //public void Post([FromBody]string value)
+        //{
+        //}
+
         // Policies are not deleted directly!
         // DELETE: api/Policy/5
         //public void Delete(int id)
         //{
         //}
+
+        private static ReportSettings CreateFromPolicySummary(PolicySummary policySummary)
+        {
+            return new ReportSettings
+            {
+                BaseUrl = policySummary.BaseUrl,
+                SharedSecret = policySummary.SharedSecret,
+                ProjectId = policySummary.ProjectId
+            };
+        }
+
+        private ReportSettings SyncReportSettingsToJira(long id, string baseUrl, string sharedSecret)
```

Comment:
clasa reportSettings contine un policy momentan, in viitor cand nu va mai fi, se va modifica - ma intereseaza sa si mearga, nu doar conceptual, for now
toata astea au fost facute la inceputul task-ului cand era hotarat sa fie un policyXml... oricand se schimba, vom modifica in alt task, nu pot sa revin asupra aceluiasi task refacand chestii de fiecare data cand apar modificari - motivul pentru care am cerut explicit sa termine Sebi refactorul